File: leetcode-submissions/0082-remove-duplicates-from-sorted-list-ii/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {

        ListNode dummy = new ListNode(0, head);
        ListNode curr = dummy;
        while(head != null)  {

            if (head.next != null && head.val == head.next.val) {
            while(head != null && head.next != null && head.val == head.next.val) {
                head = head.next;
            }
                curr.next = head.next;
            } else {
                curr = curr.next;
            }
            head = head.next;
        }


        return dummy.next;
    }
}

File: leetcode-submissions/0082-remove-duplicates-from-sorted-list-ii/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:

        if not head or not head.next:
            return head

        dummy = ListNode(val=0, next=head)
        prev, curr = dummy, head

        while curr and curr.next:

            if curr.val == curr.next.val:
                
                value = curr.val
                while curr and curr.val == value:
                    curr = curr.next
                prev.next = curr

            else:
                prev = curr
                curr = curr.next

        return dummy.next

